docs/source/tutorials/oregon.rst

```diff
@@ -0,0 +1,688 @@
+
+**Important**: When using this dataset for research or publications, appropriate citation is required as specified in the NBER data use agreement.
+
+.. code-block:: python
+
+    import numpy as np
+    import pandas as pd
+    import matplotlib.pyplot as plt
+    import os
+    from sklearn.linear_model import LinearRegression
+    from sklearn.preprocessing import LabelEncoder
+    import dte_adj
+    from dte_adj.plot import plot
+
+    # Load the Oregon Health Insurance Experiment dataset
+    base_path = "OHIE_Public_Use_Files/OHIE_Data"
+    df_descriptive = pd.read_stata(os.path.join(base_path, "oregonhie_descriptive_vars.dta"))
+    df_ed = pd.read_stata(os.path.join(base_path, "oregonhie_ed_vars.dta"))
+    df_inp = pd.read_stata(os.path.join(base_path, "oregonhie_inperson_vars.dta"))
+    df_state = pd.read_stata(os.path.join(base_path, "oregonhie_stateprograms_vars.dta"))
+
+    # Merge all datasets
+    df = (
+        df_descriptive
+        .merge(df_ed, on='person_id', how='inner')
+        .merge(df_inp, on='person_id', how='left')
+        .merge(df_state, on='person_id', how='inner')
+    )
+
+    print(f"Dataset shape: {df.shape}")
+    print(f"Average num_visit_cens_ed by enrollment:\n{df.groupby('ohp_all_ever_inperson')['num_visit_cens_ed'].mean()}")
+    print(f"Average ed_charg_tot_ed by enrollment:\n{df.groupby('ohp_all_ever_inperson')['ed_charg_tot_ed'].mean()}")
+
+    # Prepare the data for dte_adj analysis
+    # Create treatment assignment (instrumental variable): 0=Not selected, 1=Selected
+    treatment_assignment_mapping = {'Not selected': 0, 'Selected': 1}
+    df['Z'] = df['treatment'].map(treatment_assignment_mapping).astype(float).fillna(-1).astype(int)
+
+    # Create actual treatment indicator: 0=Not enrolled, 1=Enrolled, -1=Missing
+    treatment_mapping = {'NOT enrolled': 0, 'Enrolled': 1}
+    df['D'] = df['ohp_all_ever_inperson'].map(treatment_mapping).astype(float).fillna(-1).astype(int)
+
+    # Use emergency department costs and visits as outcome variables
+    df['Y_ED_CHARG_TOT_ED'] = df['ed_charg_tot_ed'].fillna(0)
+    df['Y_NUM_VISIT_CENS_ED'] = df['num_visit_cens_ed'].fillna(0)
+
+    # Create strata based on household size
+    df['strata'] = df['numhh_list']
+
+    # Create feature mappings for categorical variables
+    gender_mapping = {'Male': 0, 'Female': 1, 'Transgender F to M': 2, 'Transgender M to F': 3}
+    health_last12_mapping = {'1: Very poor': 1, '2: Poor': 2, '3: Fair': 3, '4: Good': 4, '5: Very good': 5, '6: Excellent': 6}
+    edu_mapping = {'HS diploma or GED': 0, 'Post HS, not 4-year': 1, 'Less than HS': 2, '4 year degree or more': 3}
+
+    df['age'] = 2008 - df['birthyear_list']
+    df['gender_inp'] = df['gender_inp'].map(gender_mapping).astype(float).fillna(-1).astype(int)
```

Comment:
ditto

Reply:
The reason for casting to float first is as mentioned above.
However, different from the assignments (`df['Z']`, `df['D']`) mentioned above, I think it's fine to complete missing values for these features as “unknown.” What do you think?